src/archivematicaCommon/lib/appconfig.py

```diff
@@ -39,7 +41,10 @@ def get(self, attr, default=None):
 
         attr_opts = self.attrs[attr]
         if isinstance(attr_opts, list):
-            return self.get_from_opts_list(attr, attr_opts, default=default)
+            value = self._get_from_opts_list(attr, attr_opts, default=default)
+            if attr == 'search_enabled':
+                value = self._normalize_search_enabled(value)
```

Comment:
Would it be possible if you have time to update this solution so it's not specific to given attributes? E.g. we could have a new `list` type along with `allowed_values`:

    {'section': 'MCPServer', 'option': 'search_enabled', 'type': 'list', 'allowed_values': ['aips', 'transfers']},

I'd like `appconfig.Config` to be agnostic to specific configurations as much as possible. Would that be possible? Am I asking too much? I'm not sure how much extra work would that require at this point.

Reply:
Thanks @sevein!

The problem with this particular attribute is that it should allow boolean values alongside the list and always return a set. I could try to define the normalization rules in the config mapping at the attribute level alongside the options, but that will add some complexity to the mapping and normalization code. Considering that all this (the multi-option and normalization) looks to be done only for this attribute at this point I'm not sure if it worths the effort, but I'll give it a quick try ;)

Reply:
How about something like this for the config mapping:

```
CONFIG_MAPPING = {
    'search_enabled': {
        'section': 'Dashboard',
        'options': [
            {'name': 'disable_search_indexing', 'type': 'iboolean'},
            {'name': 'search_enabled', 'type': 'boolean'},
            {'name': 'search_enabled', 'type': 'list', 'allowed_values': set(['aips', 'transfers'])},
        ],
        'normalize': [
            {
                'type': bool,
                'rules': [
                    {'value': True, 'output': set(['aips', 'transfers'])},
                    {'value': False, 'output': set()},
                ],
            },
        ],
    },
}
```